lightningd/lightningd.c

```diff
@@ -180,47 +280,85 @@ static const char *find_my_path(const tal_t *ctx, const char *argv0)
 			errx(1, "Cannot find %s in $PATH", argv0);
 	}
 
+	/*~ The caller just wants the directory we're in.
+	 *
+	 * Note the magic "take()" macro here: it annotates a pointer as "to
```

Comment:
The needs to be moved else where. See the above comment. When I want to study the lightning network and you write that knowing which directory we are in is not trivial I believe you but don't want to learn at this moment how it is being done. However you document a situation where the tal take() takes place (which I did not understand until right now) and I am not even sure if I really understood it correctly. As uggested before I would suggest a MD file which explains some examples and it could point to this function and call and say here you see a non BOLT related usecase of a standard task where it comes into action...

Reply:
Agreed.  However, I will resolve this differently: by moving the unnecessary bits into subd.c, and so this is just a simple example, with take().